tests/test_getTemplate.py

```diff
@@ -0,0 +1,265 @@
+        return exposure
+
+    def _makePatches(self, tract):
+        """Populate the patches and dataId dicts, keyed on tract id, with the
+        warps of the main exposure and minimal dataIds, respectively.
+        """
+        if debug:
+            color = ['red', 'green', 'cyan', 'yellow'][tract.tract_id]
+            point = self.exposure.wcs.skyToPixel(tract.ctr_coord)
+            # Show the tract center, colored by tract id.
+            display.dot("x", point.x, point.y, ctype=color, size=30)
+
+        # Use 5th order to minimize artifacts on the templates.
+        config = lsst.afw.math.Warper.ConfigClass()
+        config.warpingKernelName = "lanczos5"
+        warper = lsst.afw.math.Warper.fromConfig(config)
+        for patchId in range(tract.num_patches.x*tract.num_patches.y):
+            patch = tract.getPatchInfo(patchId)
+            box = patch.getOuterBBox()
+
+            if debug:
+                # Show the patch corners as patch ids, colored by tract id.
+                points = self.exposure.wcs.skyToPixel(patch.wcs.pixelToSky([lsst.geom.Point2D(x)
+                                                                           for x in box.getCorners()]))
+                for p in points:
+                    display.dot(patchId, p.x, p.y, ctype=color)
+
+            # This is mostly stolen from pipe_tasks warpAndPsfMatch, but
+            # ip_diffim cannot depend on pipe_tasks.
+            xyTransform = lsst.afw.geom.makeWcsPairTransform(self.exposure.wcs, patch.wcs)
+            warpedPsf = lsst.meas.algorithms.WarpedPsf(self.exposure.psf, xyTransform)
+            warped = warper.warpExposure(patch.wcs, self.exposure, destBBox=box)
+            warped.setPsf(warpedPsf)
+            self.patches[tract.tract_id].append(warped)
+            self.dataIds[tract.tract_id].append({"tract": tract.tract_id,
+                                                 "patch": patchId,
+                                                 "band": "a"})
+
+    def _checkMetadata(self, template, config, box, wcs, nInputs):
+        """Check that the various metadata components were set correctly.
+        """
+        expectedBox = lsst.geom.Box2I(box)
+        expectedBox.grow(config.templateBorderSize)
+        self.assertEqual(template.getBBox(), expectedBox)
+        # WCS should match our exposure, not any of the coadd tracts.
+        for tract in self.patches:
+            self.assertNotEqual(template.wcs, self.patches[tract][0].wcs)
+        self.assertEqual(template.wcs, self.exposure.wcs)
+        self.assertEqual(template.photoCalib, self.exposure.photoCalib)
+        self.assertEqual(template.getXY0(), expectedBox.getMin())
+        self.assertEqual(template.filter.bandLabel, "a")
+        self.assertEqual(template.filter.physicalLabel, "a_test")
+        self.assertEqual(template.psf.getComponentCount(), nInputs)
+        # TODO: need other things to test here?
```

Comment:
I think this can be dropped now.

Reply:
Ok. I wasn't sure if there was anything else I should be checking, especially regarding the PSF. Some tests are better than no tests, but I'd rather be as comprehensive as I could now.